server/sqlflowserver.go

```diff
@@ -55,34 +56,60 @@ func (s *Server) Run(req *pb.Request, stream pb.SQLFlow_RunServer) error {
 		}
 		defer db.Close()
 	}
-	var pr *sf.PipeReader
-	if s.enableSession == true {
-		pr = s.run(req.Sql, db, s.modelDir, req.Session)
-	} else {
-		pr = s.run(req.Sql, db, s.modelDir, nil)
+
+	// FIXME(typhoonzero): split by ; can not deal with situations like
+	// "SELECT * from mytable where col <> ';';", should be fixed.
```

Comment:
To support more accurate splitting, we need the lexer in package `github.com/sql-machine-learning/sqlflow/sql`. So we need to export the lexer? If so, I would prefer splitting on the client-side...

Reply:
One possible solution described at https://github.com/sql-machine-learning/sqlflow/pull/779: expose this function as `SplitSQLStatements(s string) []string` for the `server` package.